### PR TITLE
Remove old chat commands, add shortcuts to inline edits, groupcommands

### LINKED ISSUE #1441: remove edit feature flag
As part of removing the feature flag, we should add a "Beta" label to Edit/Document/Test (just between now and GA) and remove the old version of the test command so that users don't see two Test options:

Image

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes https://github.com/sourcegraph/jetbrains/issues/1441 
## Changes

One image is worth a thousand words:

image
image

## Test plan

1. Test if all of the bellow works:
- click on every command
- try to use shortcut for every command
- click on every command from context menu

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt [hunk 1]
```diff
@@ -299,7 +305,6 @@ private constructor(
             when (commandId) {
               CommandId.Explain -> agent.server.legacyCommandsExplain()
               CommandId.Smell -> agent.server.legacyCommandsSmell()
-              CommandId.Test -> agent.server.legacyCommandsTest()
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt:302]: pls update TESTING.md btw (at least remove the old command test desc):
https://sourcegraph.com/-/editor?remote_url=github.com%2Fsourcegraph%2Fjetbrains&branch=main&file=TESTING.md&start_row=182&start_col=0&end_row=202&end_col=7&editor=JetBrains&version=v5.5.9

File: src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -252,9 +243,12 @@
             <keyboard-shortcut first-keystroke="ctrl shift H" keymap="Mac OS X 10.5+"/>
         </action>
 
-        <action id="cody.testCodeAction"
-                class="com.sourcegraph.cody.edit.actions.TestCodeAction">
+        <action id="cody.editCodeAction"
+                class="com.sourcegraph.cody.edit.actions.EditOpenOrRetryAction"
+                description="Opens the Edit Code dialog">
```
Comment (mkondratek) [inline src/main/resources/META-INF/plugin.xml:251]: nit: IIRC, in most cases we can have titles and descriptions in the bundle

File: src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt [hunk 3]
```diff
@@ -44,48 +50,75 @@ class CommandsTabPanel(
...
+      mnemonic: Int,
+      action: () -> Unit
+  ): JButton {
     val button = JButton(displayName)
+    buttons[actionId] = button
+
+    button.layout = BorderLayout()
+    button.border = BorderFactory.createEmptyBorder(0, 5, 0, 10)
+    ActionManagerEx.getInstanceEx().getAction(actionId).shortcutSet.shortcuts.firstOrNull()?.let {
+      button.add(JLabel(KeymapUtil.getShortcutText(it)), BorderLayout.LINE_END)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt:105]: this looks great (not fully intellij-like but anyway) 🚀

**Review by mkondratek — COMMENTED** [review]
the displayed shortcuts does not work for me neither on windows nor on macos

File: src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt (hunk 3 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt:105]: Yea, it does not look native but I think discoverability might be worth it, as we have plenty of unused real estate on that buttons :)

File: src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt [hunk 4]
```diff
@@ -44,48 +50,75 @@ class CommandsTabPanel(
...
+    setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10))
 
-    CommandId.values().forEach { command -> addCommandButton(command) }
+    fun addSeparator(name: String) {
+      val separator = TitledSeparator(name)
+      separator.maximumSize = Dimension(Int.MAX_VALUE, separator.preferredSize.height)
+      add(separator)
+    }
 
+    addSeparator("Edit Commands")
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt:59]: this one should be included in the if-statement below

Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt:59]: btw, when i switch off the feature flag the section remains in the commands tab

File: src/main/resources/META-INF/plugin.xml [hunk 5]
```diff
@@ -214,47 +214,44 @@
...
             <reference ref="cody.newChat"/>
         </group>
 
         <!-- Inline editing actions -->
-        <action id="cody.editCodeAction"
-                class="com.sourcegraph.cody.edit.actions.EditCodeAction"
-                text="Edit Code"
-                description="Opens the Edit Code dialog">
+        <action id="cody.testCodeAction"
+                icon="/icons/edit/generate_test.svg"
```
Comment (mkondratek) [inline src/main/resources/META-INF/plugin.xml:233]: sadly, the latest snapshot of IntelliJ does not support the icons (yet?) 😢 

image

**Comment by taylorsperry** [comment]
Thank you, @pkukielka! Are these the right hotkeys? There's some divergence from what I captured [here](https://github.com/sourcegraph/jetbrains/issues/1477) and @danielmarquespt was going to think about whether we can make any of these combinations simpler. Let's discuss in the sync.

File: src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt (hunk 4 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt:59]: Good point!

**Comment by pkukielka** [comment]
> Thank you, @pkukielka! Are these the right hotkeys? There's some divergence from what I captured [here](https://github.com/sourcegraph/jetbrains/issues/1477) and @danielmarquespt was going to think about whether we can make any of these combinations simpler. Let's discuss in the sync.

So, the right shortcut was assigned but since looks like somehow there are two shortcuts for Edit command I was displaying the wrong one.
Fixed now.

image

**Comment by pkukielka** [comment]
> Thank you, @pkukielka! Are these the right hotkeys? There's some divergence from what I captured [here](https://github.com/sourcegraph/jetbrains/issues/1477) and @danielmarquespt was going to think about whether we can make any of these combinations simpler. Let's discuss in the sync.

Good point, it was caused by a duplicated shortcut, I think coming from some of previous plugin iterations.
Fixed now.

image

**Review by steveyegge — APPROVED** [review]

**Pull request merged**